### PR TITLE
Calcproject

### PR DESCRIPTION
**Pull request by ShanaDart73**

### CODE REVIEW

File: Calculator/script.js [hunk 1]
```diff
@@ -1,161 +1,226 @@
 
-const calculator= {
+const calculator = {
```
Comment (laurianek) [inline Calculator/script.js:2]: You should wrap the whole script into a self calling function. Example => 
`(function (){ <your whole code here> })()`

File: Calculator/script.js [hunk 2]
```diff
@@ -1,161 +1,226 @@
...
-function inputDigit(digit) {
-    const { displayValue, waitingForSecondOperand } = calculator;
-    // Overwrite 'displayValue' if the current value is '0' otherwise append to it
-    if (waitingForSecondOperand === true) {
-        calculator.displayValue = digit;
-        calculator.waitingForSecondOperand = false;
-    } else {
-        calculator.displayValue = displayValue ==='0' ? digit : displayValue + digit;
-    }
+const btn = document.querySelectorAll('.btn');
```
Comment (laurianek) [inline Calculator/script.js:15]: This clearly does something unexpected
Screenshot 2021-11-14 at 22 42 25

This queries for all the .btn class but there is only one and it is the container of all the buttons and not the buttons themselves

File: Calculator/script.js [hunk 3]
```diff
@@ -1,161 +1,226 @@
...
+
+btn.forEach(button => {
+    button.addEventListener('click', (event) => {
```
Comment (laurianek) [inline Calculator/script.js:18]: see above the `button` is not a button but the whole container

File: Calculator/script.js [hunk 4]
```diff
@@ -1,161 +1,226 @@
...
+        const { target } = event;
+        const digit = target.getAttribute('data-number');
+        const operator = target.getAttribute('data-operator');
+        const equals = target.getAttribute('data-equals');
+        const dot = target.getAttribute('data-decimal');
+        const deleteUnit = target.getAttribute('data-unit');
+        const clearDisplay = target.getAttribute('data-clearAll');
+
+        switch (operator) {
+            case '+':
```
Comment (laurianek) [inline Calculator/script.js:28]: Best practice the add a comment to say the the case falls through and is not just missing code

File: Calculator/script.js [hunk 5]
```diff
@@ -1,161 +1,226 @@
...
-        calculator.firstOperand = inputValue;
-    } else if (operator) {
-        const result = calcuate(firstOperand, inputValue, operator);
+function handleOperators(operator) {
+    const plus = calculator.displayValue.includes('+');
+    const minus = calculator.displayValue.includes('-');
+    const times = calculator.displayValue.includes('*');
+    const divide = calculator.displayValue.includes('/');
+
+    if (operator !== null) {
```
Comment (laurianek) [inline Calculator/script.js:100]: can be re-written as 

```javascript
if (operator === null) return;

if (plus … ) {
  … 
}
```

File: Calculator/script.js [hunk 6]
```diff
@@ -1,161 +1,226 @@
...
+
+        if (plus && !calculator.operatorOnOff) {
```
Comment (laurianek) [inline Calculator/script.js:102]: can be rewritten like

```
if (!calculator.operatorOnOff) {
  if (plus || minus || times || divide) {
    calculate();
  }
}
```
or better
```
if ((plus || minus || times || divide) && !calculator.operatorOnOff) {
  calculate();
}
```

File: Calculator/script.js (hunk 5 above)
Comment (ShanaDart73) [reply Calculator/script.js:100]: Thanks a lot 👍 @laurianek

**Pull request merged**